src/components/VaultRow/VaultRow.tsx

```diff
@@ -36,29 +52,65 @@ export const VaultRow: FC<VaultTableRowProps> = ({
   showEarningsColumn = false,
   ...props
 }) => {
+  const isCompounderProduct = useIsCompounderProduct(
+    props.productType ?? "compounder"
+  )
+  const compounderProps = props as CompounderVaultProps
+  const concentratorProps = props as ConcentratorVaultProps
```

Comment:
There is actually a really clean way to handle this with TypeScript using tagged unions:

Start with the type definition for `VaultRowPropsWithProduct`:

```ts
export type VaultRowPropsWithProduct =
  | ({
      productType: "compounder"
    } & CompounderVaultProps)
  | ({
      productType: "concentrator"
    } & ConcentratorVaultProps)
```

This tells TS that if `productType` is `"compounder"`, the props will be of type `CompounderVaultProps`. If `productType` is `"concentrator"`, the props will be of type `ConcentratorVaultProps`. Now, you don't need to do this casting thing -- **but you can't use this hook to check the compounder product type**. 

I'm not sure if that is because it's memoized or what, but it's really not necessary anyways. You can just do at the top of this component:

```ts
const isCompounderProduct = props.productType === "compounder"
```

Reply:
Yes, sure. If we know the type, not need anymore to check the productType this way.